cmd/protecodeExecuteScan.go

```diff
@@ -261,30 +265,30 @@ func createDockerClient(config *protecodeExecuteScanOptions) piperDocker.Downloa
 }
 
 func uploadScanOrDeclareFetch(config protecodeExecuteScanOptions, productID int, client protecode.Protecode, fileName string) int {
-	//check if the LoadExistingProduct) before returns an valid product id, than scip this
-	if !hasExisting(productID, config.VerifyOnly) {
-		if len(config.FetchURL) > 0 {
-			log.Entry().Debugf("Declare fetch url %v", config.FetchURL)
-			resultData := client.DeclareFetchURL(config.CleanupMode, config.Group, config.FetchURL)
-			productID = resultData.Result.ProductID
-		} else {
-			log.Entry().Debugf("Upload file path: %v", config.FilePath)
-			if len(config.FilePath) <= 0 {
-				log.Entry().Fatalf("There is no file path configured for upload : %v", config.FilePath)
-			}
-			pathToFile := filepath.Join(config.FilePath, fileName)
-			if !(fileExists(pathToFile)) {
-				log.Entry().Fatalf("There is no file for upload: %v", pathToFile)
-			}
-
-			combinedFileName := fileName
-			if len(config.PullRequestName) > 0 {
-				combinedFileName = fmt.Sprintf("%v_%v", config.PullRequestName, fileName)
-			}
-
-			resultData := client.UploadScanFile(config.CleanupMode, config.Group, pathToFile, combinedFileName)
-			productID = resultData.Result.ProductID
+	productID = -1
+	if len(config.FetchURL) > 0 {
+		log.Entry().Debugf("Declare fetch url %v", config.FetchURL)
+		resultData := client.DeclareFetchURL(config.CleanupMode, config.Group, config.FetchURL)
+		productID = resultData.Result.ProductID
+	} else {
+		log.Entry().Debugf("Upload file path: %v", config.FilePath)
+		if len(config.FilePath) <= 0 {
+			//TODO: bubble up error
```

Comment:
Should we open issue for this, or when should it be resolved?

Reply:
Would prefer to have an issue otherwise it will get lost.

Reply:
created https://github.com/SAP/jenkins-library/issues/2898